[enhance] better error message in Login page

wgx731 opened the pull request:
@djenriquez, `vault-ui` is really an awesome project, wish I could also help with the development, as a starting point, here is a small fix for Login page.

Thank you very much.

wgx731 committed:
[bug] remove duplicate `Error` text in Login page

Lucretius commented:
Hi there,

Glad to hear you're enjoying the project.  What you've found in the code there isn't actually a bug - the word 'error' is not duplicated - it's actually making use of an ES6 JavaScript feature known as a template string.  You can read it as follows:
`this.setState({ errorMessage: `Error: ${error}` });`

Given some object `var error = 'this is an error.'`
The errorMessage property in the state object will read: "Error: this is an error.".

Thanks for your interest in the project!

djenriquez commented:
Hi @wgx731,

We would definitely love more community support! With regards to the PR, I see what you're saying about it being repetitive since its already an error, why put `Error:`? I believe we decided for this matter that its better to be overly explicatory just to make sure the user understands that it is an error. Does that make sense?

wgx731 committed:
[enhance] better error message on Login page

wgx731 commented:
@djenriquez and @Lucretius,

It's really nice to hear back from you guys so quickly, I have updated the issue title to make the context clear, it's more like a UI enhancement rather on bug.

Here is the UI screen when I have use wrong auth to login:
<img width="1676" alt="error-screen" src="https://cloud.githubusercontent.com/assets/494702/23244018/9c1e2d1e-f9bd-11e6-82c7-ce7d4d141f04.png">

Basically, I think the problem is that `axios`'s `error` object will have a `Error: ` prefix when being put in the template string. And after looking at the [`axios` doc](https://github.com/mzabriskie/axios#handling-errors), I have changed my enhancement. Please review if it make sense this time.

Thank you very much.

msessa commented on app/components/Login/Login.jsx:
```diff
@@ -134,7 +134,15 @@ export default class Login extends React.Component {
+                } else {
+                    loginErrorMessage = error.message;
+                }
+                this.setState({ errorMessage: `Error: ${loginErrorMessage}` });
```
This a bit verbose. When possible try to use the `lodash` module as it provides a number of helpful functions.

The same code can be written as:
```
if (_.has(error, 'response.data.errors') && error.response.data.errors.length > 0)
    loginErrorMessage = _.join(error.response.data.errors, ", ");
else
    loginErrorMessage = error.message;
```

wgx731 replied:
@msessa-cotd, refactor the code based on the feedback.

Thank you very much.

wgx731 committed:
[refactor] update based on review

djenriquez reviewed (approved):
LGTM!
